realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SyncedRealmTests.kt

```diff
@@ -508,6 +509,107 @@ class SyncedRealmTests {
         }
     }
 
+    @Test
+    fun allTypes_roundTrip() {
+        val user1: User = createNewUser()
+        val config1: SyncConfiguration = createDefaultConfig(user1, partitionValue)
+
+        val primaryKeyValue = ObjectId()
+        val expectedRealmInteger = 100.toLong()
+        val expectedString = "hello world"
+        val expectedLong = 10.toLong()
+        val expectedDouble = 10.0
+        val expectedBoolean = true
+        val expectedDate = Date()
+        val expectedBinary = byteArrayOf(0, 1, 0)
+        val expectedDecimal128 = Decimal128(10)
+        val expectedObjectId = ObjectId()
+        val expectedUUID = UUID.randomUUID()
+        var expectedRealmObject = SyncDog().apply {
+            id = expectedObjectId
+        }
+        val expectedRealmList = RealmList<SyncDog>()
+        val expectedStringList = RealmList<String>("hello world 1", "hello world 2")
+        val expectedBinaryList = RealmList<ByteArray>()
```

Comment:
This list should probably contain some data?

Reply:
Indeed it needs some data.